webapp/channels/src/components/advanced_text_editor/use_groups.tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useCallback, useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {GroupSource} from '@mattermost/types/groups';
+
+import {getChannelMemberCountsByGroup} from 'mattermost-redux/actions/channels';
+import {Permissions} from 'mattermost-redux/constants';
+import {getChannel, getChannelMemberCountsByGroup as selectChannelMemberCountsByGroup} from 'mattermost-redux/selectors/entities/channels';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {getAssociatedGroupsForReferenceByMention} from 'mattermost-redux/selectors/entities/groups';
+import {isCustomGroupsEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+
+import {searchAssociatedGroupsForReference} from 'actions/views/group';
+
+import Constants from 'utils/constants';
+import {groupsMentionedInText, mentionsMinusSpecialMentionsInText} from 'utils/post_utils';
+
+import type {GlobalState} from 'types/store';
+
+const useGroups = (
```

Comment:
I tried to move all the group relate logic here.

It mainly ensures we have the groups information, and handles how we get the group mentions from the message.

I don't like the group logic in general, and probably can be improved, but that is out of the scope of this PR.

Reply:
I'm not really used to seeing code organized this way, but it makes sense to me since it helps keep everything groups-related in one place. There's a lot going on here, but it's sort of like `usePlaybooksCRUD` or some of the hooks I want to build for data manager in that way.